packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -143,17 +109,26 @@ export default function markdown({ settings, logging }: AstroPluginOptions): Plu
 				const code = escapeViteEnvReferences(`
 				import { Fragment, jsx as h } from ${JSON.stringify(astroJsxRuntimeModulePath)};
 				${layout ? `import Layout from ${JSON.stringify(layout)};` : ''}
-				${
-					settings.config.experimental.assets
-						? 'import { getConfiguredImageService } from "astro:assets";\ngetConfiguredImageService();'
-						: ''
+				${settings.config.experimental.assets ? 'import { getImage } from "astro:assets";' : ''}
+
+				export const images = {
+					${imagePaths.map(
+						(entry) =>
+							`'${entry.raw}': await getImage({src: (await import("${entry.absolute}")).default})`
+					)}
 				}
 
-				const images = {
-					${imagePaths.map((entry) => `'${entry}': await import('${entry}')`)}
+				function updateImageReferences(html) {
+					return html.replaceAll(
+						/__ASTRO_IMAGE_=\"(.+)\"/gm,
+						(full, imagePath) =>
+							\`src="\${images[imagePath].src}" \${Object.entries(images[imagePath].attributes)
+								.map((entry) => \`\${entry[0]}="\${entry[1]}"\`)
+								.join(" ")}\`
```

Comment:
This could use our internal [`spreadAttributes`](https://github.com/withastro/astro/blob/28373452503bc6ca88221ffd39a5590e015e4d71/packages/astro/src/runtime/server/index.ts#LL68C17-L68C33) function if you want to clean this up! Handles escaping quotes and sanitization.